common/primitives/json.go

```diff
@@ -56,43 +68,58 @@ func ParseJSON2Request(request string) (*JSON2Request, error) {
 	return j, nil
 }
 
+// JSON2Response is an JSON RPC response object used in factomd according to the specification for JSON RPC 2.0
+// See specification: https://www.jsonrpc.org/specification
+// Remote Proceedure Call (RPC) is a protocol for executing subroutines on a remote
 type JSON2Response struct {
-	JSONRPC string      `json:"jsonrpc"`
-	ID      interface{} `json:"id"`
-	Error   *JSONError  `json:"error,omitempty"`
-	Result  interface{} `json:"result,omitempty"`
+	JSONRPC string      `json:"jsonrpc"` // version string which MUST be "2.0" (version 1.0 didn't have this field)
+	ID      interface{} `json:"id"`      // Unique client defined ID associated with incoming request. It may be a number,
+	// string, or nil. It is used by the remote to formulate a this response object
+	// containing the same ID back to the client. If nil, remote treats request object
+	// as a notification, and that the client does not expect a response object back
+	Error  *JSONError  `json:"error,omitempty"`  // Must be present if called subroutine had an error (mutually exclusive with Result)
+	Result interface{} `json:"result,omitempty"` // Must be present if called subroutine succeeded (mutually exclusive with Error)
 }
 
-func (e *JSON2Response) JSONByte() ([]byte, error) {
-	return EncodeJSON(e)
+// JSONByte encodes object to a json []byte
+func (j *JSON2Response) JSONByte() ([]byte, error) {
+	return EncodeJSON(j)
 }
 
-func (e *JSON2Response) JSONString() (string, error) {
-	return EncodeJSONString(e)
+// JSONString encodes object to a json string with error return
+func (j *JSON2Response) JSONString() (string, error) {
+	return EncodeJSONString(j)
 }
 
-func (e *JSON2Response) String() string {
-	str, _ := e.JSONString()
+// String encodes object to a json string with no error return
+func (j *JSON2Response) String() string {
+	str, _ := j.JSONString()
 	return str
 }
 
+// NewJSON2Response returns a new JSON2Response initialized with defaults
 func NewJSON2Response() *JSON2Response {
 	j := new(JSON2Response)
 	j.JSONRPC = "2.0"
 	return j
 }
 
+// AddError sets the member Error with a new JSONError formed from the inputs
 func (j *JSON2Response) AddError(code int, message string, data interface{}) {
-	e := NewJSONError(code, message, data)
-	j.Error = e
+	err := NewJSONError(code, message, data)
```

Comment:
why this intermediary variable `err` at all then :)

Reply:
good catch, i missed that in clean up. :-)